src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_commands.c

```diff
@@ -0,0 +1,662 @@
+            os_strdup(upgrade_custom_task->custom_installer, installer);
+        }
+    }
+
+    if (!installer) {
+        if (!strcmp(agent_task->agent_info->platform, "windows")) {
+            os_strdup("upgrade.bat", installer);
+        } else {
+            os_strdup("upgrade.sh", installer);
+        }
+    }
+
+    wpk_path = basename_ex(file_path);
+
+    // lock_restart
+    if (!wm_agent_upgrade_send_lock_restart(agent_task->agent_info->agent_id)) {
+
+        // open wb
+        if (!wm_agent_upgrade_send_open(agent_task->agent_info->agent_id, wpk_path)) {
+
+            // write
+            if (!wm_agent_upgrade_send_write(agent_task->agent_info->agent_id, wpk_path, file_path, manager_configs->chunk_size)) {
+
+                // close
+                if (!wm_agent_upgrade_send_close(agent_task->agent_info->agent_id, wpk_path)) {
+
+                    // sha1
+                    result = wm_agent_upgrade_send_sha1(agent_task->agent_info->agent_id, wpk_path, file_sha1);
+                }
+            }
+        }
+    }
+
+    if (!result) {
+        result = wm_agent_upgrade_send_upgrade(agent_task->agent_info->agent_id, wpk_path, installer);
+    }
+
+    os_free(file_path);
+    os_free(file_sha1);
+    os_free(installer);
+
+    return result;
+}
+
+STATIC int wm_agent_upgrade_send_lock_restart(int agent_id) {
+    int result = OS_INVALID;
+    char *command = NULL;
+    char *response = NULL;
+
+    os_calloc(OS_MAXSTR, sizeof(char), command);
+
+    snprintf(command, OS_MAXSTR, "%.3d com lock_restart -1",agent_id);
+
+    response = wm_agent_upgrade_send_command_to_agent(command, strlen(command));
+
+    result = wm_agent_upgrade_parse_agent_response(response, NULL);
+
+    os_free(command);
+    os_free(response);
+
+    return result;
+}
+
+STATIC int wm_agent_upgrade_send_open(int agent_id, const char *wpk_file) {
+    int result = OS_INVALID;
+    char *command = NULL;
+    char *response = NULL;
+    int open_retries = 0;
+
+    os_calloc(OS_MAXSTR, sizeof(char), command);
+
+    snprintf(command, OS_MAXSTR, "%.3d com open wb %s", agent_id, wpk_file);
+
+    for (open_retries = 0; open_retries < WM_UPGRADE_WPK_OPEN_ATTEMPTS; ++open_retries) {
+        os_free(response);
+        response = wm_agent_upgrade_send_command_to_agent(command, strlen(command));
+        if (result = wm_agent_upgrade_parse_agent_response(response, NULL), !result) {
+            break;
+        }
+    }
+
+    os_free(command);
+    os_free(response);
+
+    return result;
+}
+
+STATIC int wm_agent_upgrade_send_write(int agent_id, const char *wpk_file, const char *file_path, int chunk_size) {
+    int result = OS_INVALID;
+    char *command = NULL;
+    char *response = NULL;
+    FILE *file = NULL;
+    unsigned char buffer[chunk_size];
+    size_t bytes = 0;
+    size_t command_size = 0;
+    size_t byte = 0;
+
+    os_calloc(OS_MAXSTR, sizeof(char), command);
+
+    if (file = fopen(file_path, "rb"), file) {
```

Comment:
Just like above, please separate this condition.

Reply:
Done